api/spec/requests/spree/api/checkouts_controller_spec.rb

```diff
@@ -353,7 +353,10 @@ module Spree
       end
 
       it "can apply a coupon code to an order" do
+        allow(Spree::Deprecation).to receive(:warn)
```

Comment:
Why this `allow` is needed here? Shouldn't `expect` provide both stub and expectation?

Reply:
This should be a partial mock, it can recevive warnings about anything (`firstname` usage included) but verify the expectation only on the `CouponCodesController#create` deprecation its checking.